src/algebra/order/ring.lean

```diff
@@ -512,6 +512,9 @@ calc a * b ≤ b : decidable.mul_le_of_le_one_left hb0 ha
 lemma mul_lt_one_of_nonneg_of_lt_one_right : a ≤ 1 → 0 ≤ b → b < 1 → a * b < 1 :=
 by classical; exact decidable.mul_lt_one_of_nonneg_of_lt_one_right
 
+lemma mul_lt_one (ha : a < 1) (hb' : 0 ≤ b) (hb : b < 1) : a * b < 1 :=
+mul_lt_one_of_nonneg_of_lt_one_right (le_of_lt ha) hb' hb
```

Comment:
What's the point over the previous lemma?

Reply:
You mean why add `mul_lt_one` when we already have `mul_lt_one_of_nonneg_of_lt_one_right`? Just because it's obvious that `(a < 1) -> (b < 1) -> (a * b < 1)` but `library_search` fails, and it's frustrating to have to dig around to find the required tweak to the inputs.

Reply:
In particular this came up while writing the instances `semigroup (Ico 0 1)` and `semigroup (Ioo 0 1)` that you suggested a couple of days ago.  For example, when showing that `p*q ∈ Ico 0 1` this lemma lets us write `mul_lt_one p.2.2 q.2.1 q.2.2` rather than `mul_lt_one_of_nonneg_of_lt_one_right (le_of_lt p.2.2) q.2.1 q.2.2`.

Reply:
You know that you can write `h.le` instead of `le_of_lt h`, right?

Reply:
Ah, good point.  But still, `library_search` doesn't seem to know that it can make this conversion, so it still fails to turn anything up.

Reply:
I'm not convinced that we want to write `2^n` version of each lemma where `n` is the number of times `≤` appears in it.

Reply:
Ok, I'll delete the new lemma and just leave the golf.